GeosCore/mercury_mod.F90

```diff
@@ -692,38 +724,78 @@ SUBROUTINE ChemMercury( Input_Opt,  State_Chm, State_Diag,                 &
        !--------------------------------------------------------------------
        IF ( IERR < 0 ) THEN
 
+#ifdef MODEL_GEOS
+          ! Save a backup copy of C (for GEOS-Chem in GEOS only)
+          localC    = C
+#endif
+
           ! Reset first time step and start concentrations
           ! Retry the integration with non-optimized
           ! settings
-          RCNTRL(3)  = 0e+0_fp
-          CALL Init_KPP( )
-          VAR = C(1:NVAR)
-          FIX = C(NVAR+1:NSPEC)
+          RCNTRL(3) = 0e+0_fp
+          C         = 0.0_dp
+
+          ! Update rate constants again
           CALL Update_RCONST( )
+
+          ! Integrate again
           CALL Integrate( TIN,    TOUT,    ICNTRL,                           &
                           RCNTRL, ISTATUS, RSTATE, IERR                     )
 
           !------------------------------------------------------------------
           ! Exit upon the second failure
           !------------------------------------------------------------------
           IF ( IERR < 0 ) THEN
-             WRITE(6,*) '## INTEGRATE FAILED TWICE !!! '
-             WRITE(ERRMSG,'(a,i3)') 'Integrator error code :',IERR
-             CALL ERROR_STOP(ERRMSG, 'INTEGRATE_KPP')
-          ENDIF
+             WRITE(6,     '(a   )' ) '## INTEGRATE FAILED TWICE !!! '
+             WRITE(ERRMSG,'(a,i3)' ) 'Integrator error code :', IERR
```

Comment:
Added code in mercury_mod.F90 to exit the simulation if KPP fails twice.  This is similar to what we do in fullchem_mod.F90.